controller/main.go

```diff
@@ -123,14 +123,17 @@ func (c *controller) SetPools(l log.Logger, pools *config.Pools) controllers.Syn
 	level.Debug(l).Log("event", "startUpdate", "msg", "start of config update")
 	defer level.Debug(l).Log("event", "endUpdate", "msg", "end of config update")
 
-	if pools == nil || pools.ByName == nil {
-		level.Error(l).Log("op", "setConfig", "error", "no MetalLB configuration in cluster", "msg", "configuration is missing, MetalLB will not function")
+	if pools == nil || len(pools.ByName) == 0 {
+		level.Error(l).Log("op", "setConfig", "error", fmt.Sprintf("%+v", pools), "msg", "MetalLB configuration is not present")
 		return controllers.SyncStateErrorNoRetry
+	} else {
+		level.Info(l).Log("op", "setConfig", "msg", "MetalLB configuration is present")
```

Comment:
No need for the else, as we are returning above.
Also, I find this log redundant with the "applied below, let's keep only the "applied" one. I'd also remove the dump of the pools. When dump the whole struct in debug as soon as we receive it, I think it's enough.